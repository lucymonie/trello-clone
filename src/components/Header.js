import React from 'react';

const Header = ({ title }) => (
  <header className="blue darken-3 text-white center">
    <span className="title blue-text text-lighten-3">{ title }</span>
  </header>
)


export default Header;
