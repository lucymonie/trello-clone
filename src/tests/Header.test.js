import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';

describe("Header", () => {
  it("renders consistently", () => {
    const header = shallow(<Header />);
    expect(header).toMatchSnapshot();
  });

  it("renders a title if this is passed in through props", () => {
    const title = "Title";
    const header = shallow(<Header title={title} />);
    expect(header.text()).toBe("Title");
  });
  
});
