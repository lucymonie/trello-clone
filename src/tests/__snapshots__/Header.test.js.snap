// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="title blue-text text-lighten-3"
/>,
      "className": "blue darken-3 text-white center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "title blue-text text-lighten-3",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="title blue-text text-lighten-3"
/>,
        "className": "blue darken-3 text-white center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "title blue-text text-lighten-3",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
